### DISCUSSION TITLE
umi 打包过后的体积实在太大怎么破？

### DISCUSSION THREAD

**Discussion by KaerElementye** [umijs/umi#8625] [General]
刚开始没用chainWebpack，通过cross-env观察后做了拆包配置。
但是node_modules下的 typescript 的包的体积实在太大，单个包就有3M+，还有办法拆下或者压缩以及优化之类的吗？

chainWebpack 配置：
`typescript: {
    name: "typescript",
    test: /[\\/]node_modules[\\/](typescript)[\\/]/,
    priority: -9,
    enforce: true,
  },`
拆出来的dist/typescript文件还是很大.....

**Comment by KaerElementye** [comment]
或者有什么办法去除一些这个包里我没用到的东西吗？

**Comment by xiaohuoni** [comment]
为什么会有typescript？

**Reply by KaerElementye** [reply]
因为项目中涉及到typescript的一些语法和用法。

**Reply by KaerElementye** [reply]
有办法在打包的时候仅保留我用到的部分，而舍弃其他的配置吗

**Comment by fz6m** [comment]
给个复现 demo 看看。

**Comment by Zlt1012** [comment]
我也遇到了,请问你解决了嘛？
<img width="322" alt="image" src="https://user-images.githubusercontent.com/57436734/179921919-4703358f-5ce2-4906-a598-269b89f9bf9e.png">

**Reply by KaerElementye** [reply]
你这应该还没拆包吧，用webpack的分包工具可以拆成多个文件，这样上线之后就是并发请求多个文件，首页白屏时间就短很多

**Comment by webshi** [comment]
umi4 使用 `chainWebpack` 拆包之后，拆出来的包并没有被 `index.html` 的 `script` 引入，`index.html` 中还是只引入了 `umi.js`，请问这个怎么解决呢
